app.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func checkingBrackets(line string) bool {
+	stack := make([]rune, 0)
+
+	for _, char := range line {
+		switch char {
+		case '(':
+			stack = append(stack, char)
+		case ')':
+			if stack[len(stack)-1] != '(' || len(stack) == 0 {
+				return false
+			}
+			stack = stack[:len(stack)-1]
+		}
+	}
+
+	return len(stack) == 0
+}
+
+func convertationFromPN(pn []string) (int, error) {
+	stack := make([]int, 0)
+
+	for _, symbol := range pn {
+		number, err := strconv.Atoi(symbol)
+		if err == nil {
+			stack = append(stack, number)
+		} else {
+			if len(stack) < 2 {
+				return -1, fmt.Errorf("Incorrect expression")
+			}
+
+			right := stack[len(stack)-1]
+			left := stack[len(stack)-2]
+			stack = stack[:len(stack)-2]
+
+			switch symbol {
+			case "+":
+				stack = append(stack, left+right)
+			case "-":
+				stack = append(stack, left-right)
+			case "*":
+				stack = append(stack, left*right)
+			case "/":
+				stack = append(stack, left/right)
+			}
+		}
```

Comment:
Давай все же бороться с излишней вложенностью

```suggestion
		if err == nil {
			stack = append(stack, number)
			continue
		}
		if len(stack) < 2 {
			return -1, fmt.Errorf("Incorrect expression")
		}

		right := stack[len(stack)-1]
		left := stack[len(stack)-2]
		stack = stack[:len(stack)-2]

		switch symbol {
		case "+":
			stack = append(stack, left+right)
		case "-":
			stack = append(stack, left-right)
		case "*":
			stack = append(stack, left*right)
		case "/":
			stack = append(stack, left/right)
		}
```